bitwarden_access_manager.py

```diff
@@ -0,0 +1,672 @@
+            },
+            auth=(client_id, client_secret),
+        )
+        bearer_token = response.json()["access_token"]
+        return BitwardenClient(HTTPSConnection("api.bitwarden.com"), bearer_token)
+
+    def _http_get(self, url: str) -> HTTPResponse:
+        self.connection.request(
+            method="GET",
+            url=url,
+            headers={
+                "Accept": "application/json",
+                "Authorization": f"Bearer {self.bearer_token}",
+            },
+        )
+
+        return self.connection.getresponse()
+
+    def get_group(self, id: str) -> Any:
+        return json.load(self._http_get(f"/public/groups/{id}"))
+
+    def get_groups(self) -> Iterable[Group]:
+        data = self._http_get(f"/public/groups")
+        groups = json.load(data)
+        for group in groups["data"]:
+            yield Group(
+                id=group["id"],
+                name=group["name"],
+                access_all=group["accessAll"],
+            )
+
+    def get_collection_members(self, groups: Any) -> Iterable[MemberCollectionAccess]:
+            for group in groups:
+                group_id = group["id"]
+
+                data = self._http_get(f"/public/groups/{group_id}/member-ids")
+                member_ids = json.load(data)
+
+                for member_id in member_ids:
+                    data = self._http_get(f"/public/members/{member_id}")
+                    member = json.load(data)
+                    yield MemberCollectionAccess(
+                        name=member["name"],
+                    )
+
+    def get_collection_groups(self, groups: Any) -> Iterable[GroupCollectionAccess]:
+
+            for group in groups:
+                if group["readOnly"] == True:
+                    access = "readonly"
+                else:
+                    access = "write"
+
+                yield GroupCollectionAccess(
+                    name=self.get_group(group["id"])["name"],
+                    access=access,
+                )
+
+    def get_member_collections(self, members: Tuple[MemberCollectionAccess, ...]):
+        print("JERE")
+        for member in members:
+            data = self._http_get(f"/public/members/{member.id}")
+            member = json.load(data)
+            print(member)
+
+    def get_collection(self, id: str) -> Any:
+        return json.load(self._http_get(f"/public/collections/{id}"))
+
+    def get_collections(self) -> Iterable[Collection]:
+        data = self._http_get(f"/public/collections")
+        collections = json.load(data)
+
+        for collection in collections["data"]:
+            group_accesses: Optional[Tuple[GroupCollectionAccess, ...]] = None
+            member_accesses: Optional[Tuple[MemberCollectionAccess, ...]] = None
+            collection_data = self.get_collection(collection["id"])
+
+            group_accesses_data = tuple(sorted(self.get_collection_groups(collection_data["groups"])))
+
+            if group_accesses_data:
```

Comment:
```suggestion
            if len(group_accesses_data) > 0:
```

In general I try to avoid relying on implicit boolean conversion, because some values can be falsey, and in some cases that can be unexpected. Then from the code, you can never tell if that’s a bug or intentional.

E.g.

```python
x: Optional[int]

if x:
    print(x)
else:
    print("x is missing")
    # We also get here when x = 0 ... is that a bug?
```

In this case of the tuple, we assign it directly above so it’s definitely not `None`, and I _think_ the only reason for a tuple to be falsey is when it is empty, so I think this case is fine, but to avoid any possible confusion, I would still do an explicit `len` check.

Reply:
Fair point, I should use that practice in the future !

[ebf5372ed](https://github.com/ChorusOne/github-access-manager/pull/10/commits/ebf5372ed8358b3de41f217494492f1d59f7139c)